### PR TITLE
add examples of using cas fn with coll and using loop with coll

### PR DESCRIPTION
**Pull request by the-dress-code**
Good evening. I would like to push my changes. Am I pushing these to the correct area?

### CODE REVIEW

File: src/wendy-side-car.clj [hunk 1]
```diff
@@ -3,7 +3,37 @@
 (defn cas
   [foo bar]
   (if (< foo bar)
-    [foo bar]
-    [bar foo]))
+    {:data [foo bar] :swapped? false}
+    {:data [bar foo] :swapped? true}))
```
Comment (lambduhh) [inline src/wendy-side-car.clj:8]: I love this approach, when, in doubt, returning a hash map with helpful values is always a good play.


I would like to see this function named better, and with an added doc string, though :)

File: src/wendy-side-car.clj [hunk 2]
```diff
@@ -3,7 +3,37 @@
...
+
+(defn bubble-cycle 
+  [coll]
+  (loop [result []
+         [first-int next-int & the-rest] coll
+         swap-count 0]
+
```
Comment (lambduhh) [inline src/wendy-side-car.clj]: rm this extra line

File: src/wendy-side-car.clj [hunk 3]
```diff
@@ -3,7 +3,37 @@
...
+    (let [{:keys [data swapped?]} (cas first-int next-int)
+          [x y] data
+          swap-test (if swapped? 
+                      (inc swap-count)
+                      swap-count)]
+
```
Comment (lambduhh) [inline src/wendy-side-car.clj]: rm this extra line

File: src/wendy-side-car.clj [hunk 4]
```diff
@@ -3,7 +3,37 @@
...
+         :swapped swap-test}))))
+
+(defn bubble-sort
+  [coll]
+  (loop [{:keys [result swapped]} (bubble-cycle coll)]
+    (if (> swapped 0)
+      (recur (bubble-cycle result))
+      result)))
+
+(bubble-sort '[5 3 4 2 1])
```
Comment (lambduhh) [inline src/wendy-side-car.clj:36]: remember to stick this in your "rich comment block" :) Z

I'd also love to see docstrings on all of your functions.. but actually hehe I have an idea



@patclanett why don't you pull down this branch and add a commit containing docstrings and then @the-dress-code you can review to make sure they are correct ?

File: src/wendy-side-car.clj (hunk 1 above)
Comment (the-dress-code) [reply src/wendy-side-car.clj:8]: Will do!

**Comment by patclanett** [comment]
lgtm!

File: src/wendy-side-car.clj (hunk 4 above)
Comment (lambduhh) [reply src/wendy-side-car.clj:36]: OK I see this request was misunderstood.

thank you for adding docstrings @the-dress-code 
@patclanett you lucky you got out of this one :)

**Review by lambduhh — APPROVED** [review]

**Review by patclanett — APPROVED** [review]
lgtm! x2

**Pull request merged**